packages/agent/src/app.ts

```diff
@@ -518,11 +545,37 @@ export class App {
     }
 
     const address = new URL(message.remote);
-    const client = new Hl7Client({
-      host: address.hostname,
-      port: Number.parseInt(address.port, 10),
-      encoding: address.searchParams.get('encoding') ?? undefined,
-    });
+
+    let client: Hl7Client;
+
+    if (this.hl7Clients.has(message.remote)) {
+      client = this.hl7Clients.get(message.remote) as Hl7Client;
+    } else {
+      const encoding = address.searchParams.get('encoding') ?? undefined;
+      const keepAlive = this.keepAlive;
+      client = new Hl7Client({
+        host: address.hostname,
+        port: Number.parseInt(address.port, 10),
+        encoding,
+        keepAlive,
+      });
+      this.log.info(`Client created for remote '${message.remote}'`, { keepAlive, encoding });
+
+      if (client.keepAlive) {
+        this.hl7Clients.set(message.remote, client);
```

Comment:
Mental note - `message.remote` should include host, port, and other connection details, so this is a good unique identifier 👍